File: typescript/0003_longest_substring_without_repeating_characters/0003_longest_substring_without_repeating_characters.test.ts
import { lengthOfLongestSubstring } from "./0003_longest_substring_without_repeating_characters";

test("lengthOfLongestSubstring1", () => {
  const input = "abcabcbb";
  expect(lengthOfLongestSubstring(input)).toEqual(3);
});

test("lengthOfLongestSubstring2", () => {
  const input = "bbbbb";
  expect(lengthOfLongestSubstring(input)).toEqual(1);
});

test("lengthOfLongestSubstring3", () => {
  const input = "pwwkew";
  expect(lengthOfLongestSubstring(input)).toEqual(3);
});

File: cpp/0003_longest_substring_without_repeating_characters/0003_longest_substring_without_repeating_characters_test.cc
#include "0003_longest_substring_without_repeating_characters.h"
#include <gtest/gtest.h>
#include <string>

using ::std::string;

TEST(LengthOfLongestSubstring, Example1) {
  string s = "abcabcbb";

  EXPECT_EQ(Solution().lengthOfLongestSubstring(s), 3);
}

TEST(LengthOfLongestSubstring, Example2) {
  string s = "bbbbb";

  EXPECT_EQ(Solution().lengthOfLongestSubstring(s), 1);
}

TEST(LengthOfLongestSubstring, Example3) {
  string s = "pwwkew";

  EXPECT_EQ(Solution().lengthOfLongestSubstring(s), 3);
}

TEST(LengthOfLongestSubstring, Example4) {
  string s = "";

  EXPECT_EQ(Solution().lengthOfLongestSubstring(s), 0);
}
